x/campaign/types/total_supply.go

```diff
@@ -21,3 +25,34 @@ func UpdateTotalSupply(coins, updatedCoins sdk.Coins) sdk.Coins {
 
 	return append(updatedCoins, notUpdated...).Sort()
 }
+
+// ValidateTotalSupply checks whether the total supply for each denom is within the provided total supply range
+func ValidateTotalSupply(coins sdk.Coins, supplyRange TotalSupplyRange) error {
```

Comment:
```suggestion
func ValidateCoinSupplies(coins sdk.Coins, supplyRange TotalSupplyRange) error {
```

I think this name is a bit clearer

Reply:
`TotalSupply` is our current term in the official terminology of SPN to designate the supply of tokens of a campaign's mainnet. If you have idea for renaming this should be further discussed in a call because there are many places where it would change
I believe Total Supply is a good term because it is very used in the ecosystem

Reply:
Agreed with @lubtd, that was my intention when writing the function. I'd keep it as it is tbh